### ISSUE TITLE
Group steps per scenario in test report

### ISSUE THREAD

**Issue by Hsilgos** [Tyler-Keith-Thompson/CucumberSwift#72]
Hi again. May I ask if there is a reason why all steps are listed as plain list in test results? 
I tried the following:

```
public final class CucumberTest: XCTestCase {
    override public class var defaultTestSuite: XCTestSuite {
        ....
        // Instead of this:
        // allGeneratedTests.forEach { suite.addTest($0) }
        // Call this:
        generateAlltests(suite)
        return suite
    }

    static func generateAlltests(_ rootSuite: XCTestSuite) {
        for feature in Cucumber.shared.features.taggedElements(with: Cucumber.shared.environment, askImplementor: false) {
            let className = feature.title.toClassString() + readFeatureScenarioDelimiter()

            for scenario in feature.scenarios.taggedElements(with: Cucumber.shared.environment, askImplementor: true) {
                let childSuite = XCTestSuite(name: scenario.title.toClassString())
                var tests = [XCTestCase]()
                createTestCaseFor(className: className, scenario: scenario, tests: &tests)
                tests.forEach{ childSuite.addTest($0) }
                rootSuite.addTest(childSuite)
            }
        }
    }
```
And my test reports become much nicer, see screenshot.
Does it make sense to continue to improve in this direction?

<img width="807" alt="Screenshot 2023-02-02 at 23 07 56" src="https://user-images.githubusercontent.com/4347948/216461226-b12dd493-1f75-4f12-a4ca-d2cde269b325.png">

**Comment by Tyler-Keith-Thompson** [comment]
That looks great! Feel free to submit a PR.

**Comment by Tyler-Keith-Thompson** [comment]
This is available in CucumberSwift v3.3.26
